types/smtp-client/index.d.ts

```diff
@@ -7,52 +7,71 @@
 
 export class SMTPClient {
     constructor(options?: SMTPClientOptions);
-    connect(options?: ConnectOptions): Promise<void>;
-    authPlain(options?: AuthPlainOptions): Promise<void>;
     authLogin(options?: AuthLoginOptions): Promise<void>;
-    close(options?: AuthLoginOptions): Promise<void>;
-    greet(options?: GreetOptions): Promise<void>;
+    authPlain(options?: AuthPlainOptions): Promise<void>;
+    close(options?: TimeoutOptions): Promise<void>;
+    connect(options?: ConnectOptions): Promise<void>;
+    data(data: string | Buffer, options?: DataOptions): Promise<void>;
     mail(options?: MailOptions): Promise<void>;
+    quit(options?: TimeoutOptions): Promise<void>;
     rcpt(options?: RcptOptions): Promise<void>;
-    data(data: string | Buffer, options?: DataOptions): Promise<void>;
-    quit(options?: QuitOptions): Promise<void>;
+    greet(options?: GreetOptions): Promise<void>;
+    helo(options?: HeloOptions): Promise<void>;
+    ehlo(options?: EhloOptions): Promise<void>;
+    hasExtension(extensions: string[]): boolean;
+    getDataSizeLimit(): number;
+    getAuthMechanisms(): string[];
+    parseEnhancedReplyCode(line: string): string | null;
+    parseReplyText(line: string): string;
+    noop(options?: TimeoutOptions): Promise<void>;
+    secure(options?: TimeoutOptions): Promise<void>;
+}
+
+export interface TimeoutOptions {
+    timeout?: number;
+}
+
+export interface HeloOptions extends TimeoutOptions {
+    hostname?: string;
+}
+
+export interface EhloOptions extends TimeoutOptions {
+    hostname?: string;
 }
 
 export interface SMTPClientOptions {
     host?: string;
     port?: number;
 }
+
 export interface ConnectOptions {
```

Comment:
Please extends this from `TimeoutOptions`

Reply:
Good catch. I removed it in favor of timeout option